operators/src/source/gdal_source/mod.rs

```diff
@@ -288,6 +387,7 @@ impl GdalRasterLoader {
         tile_time: TimeInterval,
     ) -> Result<RasterTile2D<T>> {
         match dataset_params {
+            // TODO: discuss if we need this check here. The metadata provider should only pass on loading infos if the query intersects the datasets bounds! And the tiling strategy should only generate tiles that intersect the querys bbox.
```

Comment:
we could ensure it in the metadataprovider and only do an debug_assert here. But I don't think we do that so might as well check it here and be safe?

Reply:
we could remove the geotransform and dataset size from the json files if we remove it here and rely on the dataset bounds.

Reply:
how does this get rid of the option?

Reply:
currently the dataset bounds are stored twice in the json file. once as bbox in the result descriptors and as geotransform and size in pixels.

Reply:
and for time this should be possible using the time bound of the dataset. if the query does not intersect the time interval it should produce empty data. At the same time, the metadata loadinginfo loading provider provider should skip empty tiles. The emtpy tile adapter will fill it anyway.